### DISCUSSION TITLE
How do I make a GitHub workflow trigger other workflows without PATs?

### DISCUSSION THREAD

**Discussion by TheEvilSkeleton** [community/community#25612] [Actions]
<p>Hi,</p>
<p>I’d like to know how I can make a workflow trigger another workflow after it gets pushed without using a personal access token (PAT). I have recently found out that I can use PATs, but I am personally not a fan of that due to manual intervention. I prefer things to be extremely easy to setup and take as less time as possible.</p>
<p>The <a href="https://github.com/TheEvilSkeleton/flatpak-remote/blob/6b851a15226bcbff770a79e9824d9921d2c8d4fc/.github/workflows/flatpak.yml" rel="noopener nofollow ugc"><code>flatpak.yml</code></a> file uses the push event so each push it starts building the package. And <a href="https://github.com/TheEvilSkeleton/flatpak-remote/blob/6b851a15226bcbff770a79e9824d9921d2c8d4fc/.github/workflows/update.yml" rel="noopener nofollow ugc">update.yml</a> is used to update its dependencies every hour and push. However, <code>update.yml</code> pushing to the repository does <em>not</em> trigger <code>flatpak.yml</code>. I’m forced to use a PAT in my repository so the <code>update.yml</code> workflow can trigger <code>flatpak.yml</code> which I find it unfriendly and time consuming.</p>
<p>Thank you.</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="TheEvilSkeleton" data-post="1" data-topic="201868">
<div class="title">
<div class="quote-controls"></div>
 TheEvilSkeleton:</div>
<blockquote>
<p>I’d like to know how I can make a workflow trigger another workflow after it gets pushed without using a personal access token (PAT).</p>
</blockquote>
</aside>
<p>The answer is unfortunately simply “you can’t”. The documentation says <a href="https://docs.github.com/en/actions/reference/authentication-in-a-workflow#using-the-github_token-in-a-workflow">the limitation is to prevent accidentally creating recursive workflows</a>, and there is no configuration option to remove it.</p>

**Reply by AndresPinerosZen** [reply]
This is just awful. They should allow people to set a depth limit on the repo GITHUB_TOKEN with a repo configuration. So I could say the token has a depth=1, which would behave like they do right now. Or I could say it has a depth=2, which means a workflow can trigger another workflow only once.

On each use they would just do `token.depth=-1`

They could establish a maximum depth limit of 15. If anyone sets that limit and then gets a huge bill, it's their fault. But please don't make people create PATs, that's such an awful security and maintainability practice.

**Reply by murari-c** [reply]
hey guys, is something related to this implemented?

**Comment by TheEvilSkeleton** [comment]
<p>Hmm… is there a way that is more convenient than using PATs?</p>
